CliFx.Tests/HelpTextSpecs.Commands.cs

```diff
@@ -110,5 +114,121 @@ private class EnvironmentVariableCommand : ICommand
+            public long Long { get; set; } = -1234567;
+
+            [CommandOption(nameof(Ulong))]
+            public ulong Ulong { get; set; } = 12345678;
+
+            [CommandOption(nameof(Float))]
+            public float Float { get; set; } = 123.4567F;
+
+            [CommandOption(nameof(Double))]
+            public double Double { get; set; } = 420.1337;
+
+            [CommandOption(nameof(Decimal))]
+            public decimal Decimal { get; set; } = 1337.420M;
+
+            // Standardizing on the US culture for the test so that our unit tests are consistent across cultures.
+            [CommandOption(nameof(DateTime))]
+            public DateTime DateTime { get; set; } = DateTime.Parse("Apr 20, 2020", Culture);
+
+            // Standardizing on the US culture for the test so that our unit tests are consistent across cultures.
+            [CommandOption(nameof(DateTimeOffset))]
+            public DateTimeOffset DateTimeOffset { get; set; } = DateTimeOffset.Parse("05/01/2008 +1:00", Culture);
+
+            [CommandOption(nameof(TimeSpan))]
+            public TimeSpan TimeSpan { get; set; } = TimeSpan.FromMinutes(123);
+
+            public enum TestEnum { Value1, Value2, Value3 };
+
+            [CommandOption(nameof(CustomEnum))]
+            public TestEnum CustomEnum { get; set; } = TestEnum.Value2;
+
+            [CommandOption(nameof(IntNullable))]
+            public int? IntNullable { get; set; } = 1337;
+
+            [CommandOption(nameof(CustomEnumNullable))]
+            public TestEnum? CustomEnumNullable { get; set; } = TestEnum.Value2;
+
+            [CommandOption(nameof(TimeSpanNullable))]
+            public TimeSpan? TimeSpanNullable { get; set; } = TimeSpan.FromMinutes(234);
+
+            [CommandOption(nameof(ObjectArray))]
+            public object[]? ObjectArray { get; set; } = new object[] { "123", 4, 3.14 };
+
+            [CommandOption(nameof(StringArray))]
+            public string[]? StringArray { get; set; } = new[] { "foo", "bar", "baz" };
+
+            [CommandOption(nameof(IntArray))]
+            public int[]? IntArray { get; set; } = new[] { 1, 2, 3 };
+
+            [CommandOption(nameof(CustomEnumArray))]
+            public TestEnum[]? CustomEnumArray { get; set; } = new[] { TestEnum.Value1, TestEnum.Value3 };
+
+            [CommandOption(nameof(IntNullableArray))]
+            public int?[]? IntNullableArray { get; set; } = new int?[] { 2, 3, 4, null, 5 };
+
+            [CommandOption(nameof(EnumerableNullable))]
+            public IEnumerable? EnumerableNullable { get; set; } = Enumerable.Repeat("foo", 3);
+
+            [CommandOption(nameof(StringEnumerable))]
+            public IEnumerable<string>? StringEnumerable { get; set; } = Enumerable.Repeat("bar", 3);
+
+            [CommandOption(nameof(StringReadOnlyList))]
+            public IReadOnlyList<string>? StringReadOnlyList { get; set; } = new List<string>() { "foo", "bar", "baz" }.AsReadOnly();
```

Comment:
`new List<string>() { "foo", "bar", "baz" }.AsReadOnly();` can just be replaced with `new[] {"foo", "bar", "baz"}`

Reply:
Woah, didn't know `T[]` is implicitly convertible to `IReadOnlyList<T>`. Fixed.